design/yaml-principles.md

```diff
@@ -0,0 +1,20 @@
+# YAML language principles
+
+Here are the principles we're using to evaluate YAML suggestions, especially when they affect the base syntax.
+These are *very* rough in priority order.
+
+1. Make simple things simple, complex things possible.
```

Comment:
examples we used for easy, medium, complex scenarios were: single-job (should be simple to describe in yaml), multi-config (medium), and job graph (complex).